src/components/App/App.js

```diff
@@ -34,6 +35,7 @@ class App extends React.Component {
 
   clearSelection = () => {
     this.setState({selectedDetails: null})
+    this.componentDidMount()
```

Comment:
should we make it a method of variable in componentDidMount then call that instead?